filter.go

```diff
@@ -8,21 +8,34 @@ import (
 	manet "github.com/multiformats/go-multiaddr-net"
 )
 
+type FilterEntry struct {
+	f      *net.IPNet
+	reject bool
+}
+
 type Filters struct {
-	mu      sync.RWMutex
-	filters map[string]*net.IPNet
+	mu            sync.RWMutex
+	filterDefault bool
+	filters       []*FilterEntry
```

Comment:
What happens when there are to entries for the same IPNet? Seems behaviour now depends on how search is implemented.

Why not using `map[string]*FilterEntry ? Also simplifies searching and deleting.